native-schema-registry/csharp/AWSGsrSerDe/AWSGsrSerDe/deserializer/avro/AvroDeserializer.cs

```diff
@@ -0,0 +1,85 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
+using System.Reflection.Metadata;
+using System.Runtime.Caching;
+using Avro;
+using Avro.Generic;
+using Avro.IO;
+using Avro.Specific;
+using AWSGsrSerDe.common;
+using AWSGsrSerDe.serializer.avro;
+
+namespace AWSGsrSerDe.deserializer.avro
+{
+    public class AvroDeserializer : IDataFormatDeserializer
+    {
+        private readonly MemoryCache _datumReaderCache = new MemoryCache("datum_reader_cache");
+        private readonly CacheItemPolicy _cacheItemPolicy = new CacheItemPolicy();
+        private AvroRecordType _avroRecordType;
+
+        public long CacheSize => _datumReaderCache.GetCount();
+
+        private AvroDeserializer()
+        {
+        }
+
+        public AvroDeserializer(GlueSchemaRegistryConfiguration configs)
+            : this()
+        {
+            Configure(configs);
+        }
+
+        public void Configure(GlueSchemaRegistryConfiguration configs)
+        {
+            _avroRecordType = configs.AvroRecordType;
+            var cacheItemExpirationTime = configs.CacheItemExpirationTime;
+            _cacheItemPolicy.SlidingExpiration = TimeSpan.FromSeconds(cacheItemExpirationTime);
+        }
+
+
+        public object Deserialize([NotNull] byte[] data, [NotNull] GlueSchemaRegistrySchema schema)
+        {
+            try
+            {
+                var schemaDefinition = schema.SchemaDef;
+                Debug.WriteLine($"Length of actual message: {data.Length}");
```

Comment:
Is this required? Logging here can cause lot of messages which are not useful. General structure for logging:

1. Log during initialization. 
2. We don't log in heavily used methods like Serialize / Deserialize.
3. Log warnings.

Reply:
This is definitely not required, I added this because we also have debug output in our Java code in our Avro serializer. I will remove it for now.